### PR TITLE
[feat]: [CDS-76868] : added api to batch migrate to override v2

### PR DESCRIPTION
**Pull request by tchaurasiya**
## Describe your changes

Combined migration and setting enabling APIs on condition of successful migration

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java [hunk 1]
```diff
@@ -374,6 +378,56 @@ public class ServiceOverridesResource {
...
+      responses =
+      {
+        @io.swagger.v3.oas.annotations.responses.
+        ApiResponse(responseCode = "default", description = "Returns Override Migration Details")
+      })
+  public ResponseDTO<List<ServiceOverrideMigrationResponseDTO>>
+  migrateAndEnableServiceOverrideV2(
+      @RequestBody(required = true, description = "Details of the Service Override to be updated", content = {
+        @Content(examples = @ExampleObject(name = "Create", summary = "Sample Service Override update request",
+                     value = DocumentationConstants.SERVICE_OVERRIDE_V2_REQUEST_DTO,
```
Comment (lovish1999) [inline 120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java:395]: I think we should change the content & request body value ? Its related to migration...

**Review by lovish1999 — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java (hunk 1 above)
Comment (tchaurasiya) [reply 120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java]: Done.

**Comment by tchaurasiya** [comment]
trigger uts

**Comment by tchaurasiya** [comment]
trigger messagecheck

**Review by ivan-83 — APPROVED** [review]
LGTM
Just to confirm, if ServiceOverrides to v2 migration fails for some accounts we will get the list of accounts in List<ServiceOverrideMigrationResponseDTO> response with details for which account migration failed and we will rerun API again.
Just make sure that serviceOverrideV2MigrationService.migrateToV2 logs enough details that we can conclude the reason why it failed.
Migration operations should be Idempotent.

**Comment by tchaurasiya** [comment]
trigger review

**Pull request merged**